src/liboslexec/splineimpl.h

```diff
@@ -43,45 +47,100 @@ namespace Spline {
 
 struct SplineBasis {
    int      basis_step;
-   Matrix44 basis;
+   float    basis[4][4];
 };
 
 // ========================================================
 //
 // Interpolation bases for splines
 //
+// The order here is very important for the SplineInterp::create
+// constructor below. Any additional modes should be added to
+// the end, or SplineInterp::create updated as well.
+//
 // ========================================================
 
-static const int kNumSplineTypes = 6;
-static const int kConstantSpline = kNumSplineTypes - 1;
-static Spline::SplineBasis gBasisSet[kNumSplineTypes] = {
-   { 1, Matrix44( (-1.0f/2.0f),  ( 3.0f/2.0f), (-3.0f/2.0f), ( 1.0f/2.0f),
-                  ( 2.0f/2.0f),  (-5.0f/2.0f), ( 4.0f/2.0f), (-1.0f/2.0f),
-                  (-1.0f/2.0f),  ( 0.0f/2.0f), ( 1.0f/2.0f), ( 0.0f/2.0f),
-                  ( 0.0f/2.0f),  ( 2.0f/2.0f), ( 0.0f/2.0f), ( 0.0f/2.0f))  },
-   { 3, Matrix44(  -1,  3, -3,  1,
-                    3, -6,  3,  0,
-                   -3,  3,  0,  0,
-                    1,  0,  0,  0) }, 
-   { 1, Matrix44( (-1.0f/6.0f), ( 3.0f/6.0f),  (-3.0f/6.0f),  (1.0f/6.0f),
-                  ( 3.0f/6.0f), (-6.0f/6.0f),  ( 3.0f/6.0f),  (0.0f/6.0f),
-                  (-3.0f/6.0f), ( 0.0f/6.0f),  ( 3.0f/6.0f),  (0.0f/6.0f),
-                  ( 1.0f/6.0f), ( 4.0f/6.0f),  ( 1.0f/6.0f),  (0.0f/6.0f)) },
-   { 2, Matrix44(  2,  1, -2,  1,
-                  -3, -2,  3, -1,
-                   0,  1,  0,  0,
-                   1,  0,  0,  0) },
-   { 1, Matrix44(  0,  0,  0,  0,
-                   0,  0,  0,  0,
-                   0, -1,  1,  0,
-                   0,  1,  0,  0) },
-   { 1, Matrix44(0.0f) }  // special marker for constant
+enum {
+    kCatmullRom,
+    kBezier,
+    kBSpline,
+    kHermite,
+    kLinear,
+    kConstant,
+    kNumSplineTypes
 };
 
+#ifdef __CUDACC__
+    rtBuffer<SplineBasis> gBasisSet;
```

Comment:
You might consider making this a regular global variable rather than an OptiX Buffer, to avoid making OptiX a dependency of ``liboslexec``. It will bloat the size of the compiled code somewhat, but I think that's a worthwhile tradeoff.

If you look in ``gabornoise.cpp``, I added an ``OSL_DEVICE`` decorator to a couple of global float constants which is defined as ``__device__`` when compiling for a CUDA target. A similar pattern might work here.